subprojects/launcher/src/main/java/org/gradle/launcher/daemon/client/DaemonClient.java

```diff
@@ -167,13 +167,16 @@ protected BuildActionResult executeBuild(Build build, DaemonClientConnection con
             result = connection.receive();
```

Comment:
❓ Receive will return null only if the underlying object input stream receives EOF. Shouldn't we also capture some or all exception types here?

Reply:
The other types of exceptions are either immediate failures (that cause the client to build to start) or are already caught in a way to cause us to retry, I believe.

Reply:
👍